File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand_1, strand_2)
    guard(strand_1, strand_2)
    strand_1.chars.zip(strand_2.chars).reject { |item| item[0] == item[1] }.length
  end

  def self.guard(a, b)
    unless a.length == b.length
      raise ArgumentError.new("Strands must be of the same length")
    end
  end
end

File: javascript/hamming/hamming.js
var helper = {
  same: function (a, b, index) {
    return a.split('')[index] === b.split('')[index];
  },
  guard: function (a, b) {
    if (a.length  != b.length) {
      throw 'DNA strands must be of equal length.';
    }
  }
};

module.exports = {
  compute: function (a, b) {
    helper.guard(a, b);

    var different = 0;
    for(var i = 0; i < a.length; i++) {
      if (!helper.same(a, b, i)) {
        different += 1;
      }
    }
    return different;
  }
};
